tools/tesseratos/src/tesseratos/scene_editor/plugin.cpp

```diff
@@ -298,36 +330,47 @@ static void showSceneHierarchy(SceneInfo& scene, Commands& cmds, EntitySelector&
     if (nodeOpen)
     {
         // Add entity to current scene (needs to be root, not a sub scene)
+
         showSceneEntities(scene.entities, scene, selector, cmds, hierarchyDepth);
         if (hierarchyDepth == 0)
         {
             ImGui::Separator();
             if (ImGui::Button("Add Scene"))
             {
-                // TODO: actually add a sub scene - should add new scene to state.imports
-                std::string sceneName = scene.name + "scene" + std::to_string(scene.scenes.size() + 1);
+                ImGui::OpenPopup("Select a scene");
+            }
+            if (tesseratos::assetSelectionPopup("Select a scene", bufferasset,
+                                                cubos::core::reflection::reflect<Scene>(), assets))
+            {
+                auto* asset = new Asset<Scene>;
```

Comment:
Same here, do you really need to allocate this with `new`? Wouldn't a normal variable be enough?